### DISCUSSION TITLE
Refactor to use the C++ libraries

### DISCUSSION THREAD

**Discussion by LTLA** [LTLA/SingleR#221] [Ideas]
@dtm2451 @j-andrews7 @friedue

With the completion of https://github.com/LTLA/singlepp, I can finally fulfill the dream of separating **SingleR**'s core compute from its "other" bits (plotting, other convenience functions). This has some moderate efficiency improvements - IIRC the fine-tuning is about two times faster - but the main appeal is that it should be much easier to maintain **SingleR** without requiring knowledge of its C++ components, which are now managed and tested separately.

To that end, I am looking for volunteers to take over the official maintainer role for **SingleR**. I'll do the necessary refactoring to insert **singlepp** into the package; I can also teach you (enough) C++ to understand what's going on at the R/C++ boundary.

**Comment by LTLA** [comment]
See #224 for ongoing work in this direction.

**Comment by LTLA** [comment]
So, it's ready. Any volunteers for maintainership? I'll throw in https://github.com/LTLA/SingleRBook to sweeten the pot.

**Reply by friedue** [reply]
Does "official maintainer role" mean "answer to github issues" and "ping LTLA when things go awry"?

**Reply by LTLA** [reply]
Pretty much. And dealing with questions on the support site. Plus any relevant Slack activity.

Most ideal case would be someone who's using **SingleR** to some extent, so they can dogfood the functions and make necessary adjustments. I don't analyze a lot of data these days, so it's hard for me to figure out whether something is user-friendly. Or even biologically correct, for that matter.

I'm happy to take care of all the C++ code - I consider that a separate project anyway, and handing that off would be a little too much to ask. (Unless someone wants to upskill and learn that kind of stuff, in which case I'd be happy to be a mentor.) It's just all the little usability/quality-of-life stuff in the package that I don't deal with anymore.

(Related is maintainership of **celldex**, for similar reasons.)

**Comment by friedue** [comment]
> I don't analyze a lot of data these days.

Same here, sadly.

Are you up for it,  @j-andrews7, @dtm2451 ? Otherwise we can ask in the biostars and bioC slack channels

**Reply by j-andrews7** [reply]
I do still use it occasion, but I am really limited on time. I'd be more willing to tag-team it with someone. Stuff like [this](https://support.bioconductor.org/p/9147816/) is really going to be beyond my ability to troubleshoot very effectively or quickly.
